### PR TITLE
UI to Auto Populate Partner Name on BU Id

### PR DESCRIPTION
**Pull request by KMayuri25**

### CODE REVIEW

File: src/app/partner/create-partner/create-partner.component.html [hunk 1]
```diff
@@ -28,20 +28,16 @@ <h1 class="page-title">Create New Registered Partner</h1>
...
             </div>
           </div>
           <div class="field col-sm-6">
             <div class="right-textbox-top pull-right">
               <label class="lbl-size  " translate="tenantUsersTab.firstName">BUSINESS UNIT</label>
               <span class="required-star">*</span>
-              <div class="item">
-                <input id = 'business_unit' type="text" class="form-control" placeholder="Business Unit" formControlName="business_unit" (click)="onClickTextBox()"  [ngClass]="{'is-invalid': control.partner_name.errors && registerForm.controls['partner_name'].touched  }"/>
+              <div tabindex="-1" class="item">
+                <input  id = 'business_unit' type="text" class="form-control" placeholder="Business Unit" formControlName="business_unit" (focusout)="onClickPartner()"  [ngClass]="{'is-invalid': control.partner_name.errors && registerForm.controls['partner_name'].touched  }"/>
```
Comment (Susmith-roy-incontact) [inline src/app/partner/create-partner/create-partner.component.html]: rename the method name to onFocusOutPartner

File: src/app/partner/create-partner/create-partner.component.html [hunk 2]
```diff
@@ -28,20 +28,16 @@ <h1 class="page-title">Create New Registered Partner</h1>
...
               </div>
             </div>
           </div>
           <div class="field col-sm-6">
             <div class="right-textbox-top pull-right">
               <label class="lbl-size  " translate="tenantUsersTab.firstName">BUSINESS UNIT</label>
               <span class="required-star">*</span>
-              <div class="item">
-                <input id = 'business_unit' type="text" class="form-control" placeholder="Business Unit" formControlName="business_unit" (click)="onClickTextBox()"  [ngClass]="{'is-invalid': control.partner_name.errors && registerForm.controls['partner_name'].touched  }"/>
+              <div tabindex="-1" class="item">
```
Comment (Susmith-roy-incontact) [inline src/app/partner/create-partner/create-partner.component.html]: why we need tabIndex = "-1"

File: src/app/partner/create-partner/create-partner.component.ts [hunk 3]
```diff
@@ -50,12 +54,14 @@ export class CreatePartnerComponent implements OnInit {
   };
 
   buildForm() {
+
+   this.businessUnitData = JSON.parse(sessionStorage.getItem('PartnerNameData') || '{}');
```
Comment (Susmith-roy-incontact) [inline src/app/partner/create-partner/create-partner.component.ts]: use constat for SessionStorage parameters

File: src/app/partner/create-partner/create-partner.component.ts [hunk 4]
```diff
@@ -70,6 +76,41 @@ export class CreatePartnerComponent implements OnInit {
     ctrl.setValue(userCategory);
   }
 
+  onClickPartner() {
+    this.showSpinner = true;
+    const businessUnit = this.control.business_unit.value;
+    this.partnerService.getBusinessUnit(businessUnit).subscribe((data: BusinessUnitName) => {
+      sessionStorage.setItem('PartnerNameData', JSON.stringify(data));
+      console.log(data);
```
Comment (Susmith-roy-incontact) [inline src/app/partner/create-partner/create-partner.component.ts]: remove

File: src/app/partner/create-partner/create-partner.component.ts [hunk 5]
```diff
@@ -70,6 +76,41 @@ export class CreatePartnerComponent implements OnInit {
...
+            this.underlinedError = true;
+          }  else {
+            this.showErrorMsg = true;
+          }
+        }
+      } else {
+        this.showErrorMsg = true;
+        this.msg = errorPartnerName.statusText;
+      }
+      sessionStorage.setItem('ErrorMsg', JSON.stringify(this.msg));
```
Comment (Susmith-roy-incontact) [inline src/app/partner/create-partner/create-partner.component.ts]: use Session storage constants

File: src/app/partner/create-partner/create-partner.component.ts (hunk 4 above)
Comment (pooja-koul-incontact) [inline src/app/partner/create-partner/create-partner.component.ts]: remove logger

File: src/app/partner/service/partner.service.ts [hunk 6]
```diff
@@ -33,16 +34,19 @@ export class PartnerService {
...
-    return this.http.get('/psa/v1/partners');
+    return this.http.get(PSAAPIRoutes.GET_ALL_PARTNERS_URI);
   }
   delete = function(uuid) {
-    return this.http.delete('/psa/v1/partner' + '/' + `${uuid}` );
+    return this.http.delete(PSAAPIRoutes.PARTNER_URI + '/' + `${uuid}` );
   };
   update = function(uuid, partnerObject) {
-    return this.http.put( '/psa/v1/partner' + '/' + `${uuid}` , partnerObject);
+    return this.http.put( PSAAPIRoutes.PARTNER_URI + '/' + `${uuid}` , partnerObject);
```
Comment (pooja-koul-incontact) [inline src/app/partner/service/partner.service.ts]: concatenate / in the App route constant file

File: e2e/src/Partner/Partner.e2e-spec.ts [hunk 7]
```diff
@@ -634,6 +639,9 @@ describe('Update partner test cases for Fuze', () => {
     partnerPage = new PartnerPage();
   });
   afterAll(() => {
+    using(seleniumJsFlavours.PartnerFuze, function (data) {
+      testUtilsNoUI.deletePartner(data.BUID);
```
Comment (pooja-koul-incontact) [inline e2e/src/Partner/Partner.e2e-spec.ts]: don't delete again and again. Delete at the end only once, when all  the scripts are done executing.

File: e2e/src/Partner/Partner.e2e-spec.ts [hunk 8]
```diff
@@ -194,7 +197,9 @@ describe('Update partner test cases', () => {
     partnerPage = new PartnerPage();
   });
   afterAll(() => {
-    testUtilsNoUI.deletePartner();
+    using(seleniumJsFlavours.PartnerGoku, function (data) {
+      testUtilsNoUI.deletePartner(data.BUID);
+    });
     testUtils.presencesynclogout();
```
Comment (pooja-koul-incontact) [inline e2e/src/Partner/Partner.e2e-spec.ts]: similarly, do not login/logout again and again, only once after complete partner suite execution, Will save a lot of time.

**Pull request merged**